st2common/st2common/openapi.yaml

```diff
@@ -0,0 +1,3781 @@
+          items:
+            type: string
+        - name: name
+          in: query
+          description: Action name filter
+          type: string
+        - name: pack
+          in: query
+          description: Action pack name filter
+          type: string
+      responses:
+        '200':
+          description: List of actions
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Action'
+          examples:
+            application/json:
+              ref: 'core.local'
+              # and stuff
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    post:
+      operationId: st2api.controllers.v1.actions:actions_controller.post
+      description: |
+        Create a new action.
+      parameters:
+        - name: action
+          in: body
+          description: Action content
+          schema:
+            $ref: '#/definitions/ActionCreateRequest'
+      x-parameters:
+        - name: user
+          in: context
+          x-as: requester_user
+          description: User running the action
+      responses:
+        '200':
+          description: Single action being created
+          schema:
+            $ref: '#/definitions/Action'
+          examples:
+            application/json:
+              ref: 'core.local'
+              # and stuff
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+  /api/v1/actions/{ref_or_id}:
+    get:
+      operationId: st2api.controllers.v1.actions:actions_controller.get_one
+      description: |
+        Get one action.
+      parameters:
+        - name: ref_or_id
+          in: path
+          description: Entity reference or id
+          type: string
+          required: true
+      x-parameters:
+        - name: user
+          in: context
+          x-as: requester_user
+          description: User running the action
+      responses:
+        '200':
+          description: Action requested
+          schema:
+            $ref: '#/definitions/Action'
+          examples:
+            application/json:
+              ref: 'core.local'
+              # and stuff
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    put:
+      operationId: st2api.controllers.v1.actions:actions_controller.put
+      description: |
+        Update an action.
+      parameters:
+        - name: ref_or_id
+          in: path
+          description: Entity reference or id
+          type: string
+          required: true
+        - name: action
+          in: body
+          description: Action content
+          schema:
+            $ref: '#/definitions/ActionUpdateRequest'
+      x-parameters:
+        - name: user
+          in: context
+          x-as: requester_user
+          description: User running the action
+      responses:
+        '200':
```

Comment:
Once again, we *could* define every single possible error for the particular endpoint or we could use an umbrella `default` scheme to signify any possible error since we stick to the same response scheme for them anyway (an object with "faultstring" property). It's up to us. I personally don't think it's worth the hassle to do it now, maybe in the release or two when we populate the rest of the file with proper descriptions or get this requested by users.